app/search_builders/collection_search_builder.rb

```diff
@@ -9,9 +11,9 @@ def include_item_ids(solr_parameters)
     solr_parameters[:fq] << "hasCollectionMember_ssim:#{item.id}"
   end
 
-  def initialize(item, processor_chain, scope)
-    super(processor_chain, scope)
-    @item = item
+  def include_contained_files(solr_parameters)
+    solr_parameters[:fq] ||= []
+    solr_parameters[:fq] << "{!join from=generic_files_tesim to=id}{!join from=hasCollectionMember_ssim to=id}id:#{collection.id}"
```

Comment:
`_tesim` would be the wrong kind of field to store an ID in.  We don't want english text stemming on IDs.  (This kind of imprecision is common in our legacy hydra apps, too.)  As for whether the joins do exactly what you want them to in this context, I don't know enough about what is going into those fields to say.